src/main/java/com/google/sps/daos/DatabaseRoomDao.java

```diff
@@ -0,0 +1,75 @@
+package com.google.sps.daos;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.cloud.firestore.SetOptions;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firestore.v1.StructuredQuery.CollectionSelector;
+
+public class DatabaseRoomDao implements RoomDao {
+
+  private DatabaseReference roomsDBReference;
+  private Firestore firestoreDb;
+  private String actualRoom;
+  private String gameId;
+
+  public DatabaseRoomDao(Firestore firestoreDb) {
+    this.firestoreDb = firestoreDb;
+  }
+  
+  @Override
+  public String createNewRoom(String gameId) {
```

Comment:
Maybe this should take a `Room` instance as its argument. `public Room create(Room room);`

Reply:
Oh yeah true let me do it

Reply:
Can you look the changes ?? I created the room inside the createNewRoom method, instead of passing it as an argument, is it the same or is it better this way?

Reply:
It can work this way. My reasoning for passing a `Room` instance here is to keep the method signature more flexible. `Room` can have fields added/removed in the future but the DAO wouldn't have to change. If gameId is the only bit of info ever needed to construct a new game, then this will work.